### PR TITLE
Add category pages

### LINKED ISSUE #120: Design for a blog page listing posts by blog tag/category
We have a blog page design for filtering posts by author (e.g. https://kedro.org/blog/author/jo-stichbury) but we also need one for filtering by category e.g. "kedro releases" or "experiment tracking" to list all posts on a particular topic.

This could be very similar to the author page, or perhaps more involved to the extent that it allows for a pair of "featured" articles that are deemed the most influential/key on that particular tag.

### PR DESCRIPTION
**Pull request by tynandebold**
## Description

Resolves #120 

## Development notes

I've added the category pages. You can reach them by clicking on the category/tag above each post snippet as seen here:

https://github.com/kedro-org/kedro-website/assets/16869061/33f8b195-43f3-49ed-adf4-57f6013ccd18

## QA notes

I haven't done any nice design work or added anything we didn't already have, given that our design team is at capacity. I thought we could add this in without really changing anything and then come back at a later date with some updated UI.

Take a look at the demo URL and see what you think.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Added tests to cover my changes, where applicable

### CODE REVIEW

File: styles/_extends.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+.postByAuthorAndCategory {
+  @include text-xl();
+
+  margin: 0;
+  padding: 0 2rem 2rem;
+
+  @include breakpoint(small) {
+    padding: 0;
+    margin: 0;
```
Comment (ravi-kumar-pilla) [inline styles/_extends.scss:17]: [Nit] Can we have the alphabetic order ?

**Review by ravi-kumar-pilla — APPROVED** [review]
This might not be related to the ticket, but on hovering the blog posts, an image appears as a tool tip which clouds the text. I see the official website also has this. If this is the expected design, please ignore it.
 
image

**Comment by tynandebold** [comment]
> This might not be related to the ticket, but on hovering the blog posts, an image appears as a tool tip which clouds the text. I see the official website also has this. If this is the expected design, please ignore it.
> 
> image

This is expected.

**Comment by stichbury** [comment]
Thank you for this @tynandebold I was so pleased to preview it and find it even supports multiple categories! So the one that has databricks and datasets offers both. Awesome stuff. Thank you so much ❤️

**Review by stichbury — APPROVED** [review]
💯

**Comment by tynandebold** [comment]
> Thank you for this @tynandebold I was so pleased to preview it and find it even supports multiple categories! So the one that has databricks and datasets offers both. Awesome stuff. Thank you so much ❤️

Thanks, Jo. Really appreciate that 💕

**Pull request merged**